crates/libs/core/src/imp/heap.rs

```diff
@@ -32,5 +42,19 @@ pub fn heap_alloc(bytes: usize) -> crate::Result<*mut c_void> {
 ///
 /// `ptr` must be a valid pointer to memory allocated by `HeapAlloc` or `HeapReAlloc`
 pub unsafe fn heap_free(ptr: *mut c_void) {
-    HeapFree(GetProcessHeap(), 0, ptr);
+    #[cfg(windows)]
+    {
+        HeapFree(GetProcessHeap(), 0, ptr);
+    }
+
+    #[cfg(not(windows))]
+    {
+        extern "C" {
+            fn free(ptr: *mut c_void);
+        }
+
+        if !ptr.is_null() {
+            free(ptr);
+        }
```

Comment:
Optional: `free` already handles the null case so could skip the extra check here. I could also think up a niche case where `free` is hooked and is expecting the pass through.

Reply:
Sure, that's fine.  I'll make the change.